src/components/ChatEntry.js

```diff
@@ -1,22 +1,35 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
+
+const ChatEntry = ({ updateChat, ...message }) => {
+  const onLikeClick = () => {
+    updateChat({ ...message, liked: !message.liked });
```

Comment:
Nothing wrong with this approach--sending the message with the updated `liked` value--to the update function, but a more common pattern would be to handle that in that update function itself.